logstreams/src/main/java/io/zeebe/logstreams/state/SnapshotStorage.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.logstreams.state;
+
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * Provides a layer of abstraction between concrete snapshot stores (i.e. Atomix's or any Raft
+ * related snapshot store) and the logstream abstraction. Should be removed as part of a refactoring
+ * to refine/define the Engine/Logstream abstractions.
+ */
+public interface SnapshotStorage extends AutoCloseable {
+
+  /**
+   * Returns an existing, temporary working directory for a snapshot using the position as ID. This
+   * method is non-deterministic, as it returns a directory with the current timestamp as part of
+   * the name, and therefore should be cached if it needs to be reused.
+   *
+   * @param snapshotPosition the position to use
+   * @return a path to an existing directory
+   */
+  Snapshot getPendingSnapshotFor(long snapshotPosition);
```

Comment:
It returns an snapshot  not an directory right?

Reply:
Correct

Reply:
Then the doc is wrong

Reply:
Ah, my bad :see_no_evil: